src/lightning/fabric/utilities/logger.py

```diff
@@ -88,8 +89,11 @@ def _flatten_dict(params: MutableMapping[Any, Any], delimiter: str = "/", parent
     result: Dict[str, Any] = {}
     for k, v in params.items():
         new_key = parent_key + delimiter + str(k) if parent_key else str(k)
-        if isinstance(v, Namespace):
+        if is_dataclass(v):
+            v = asdict(v)
```

Comment:
Do you know, does a dataclass match the type `MutableMapping[Any, Any]` as annotated in the function?

Reply:
the type of `params` is `MutableMapping[Any, Any]`, thus the type of `v` is `Any`, so a dataclass does match the type